src/test/java/com/cag/cagbackendapi/integration/UserControllerIntegrationTests.kt

```diff
@@ -0,0 +1,82 @@
+package com.cag.cagbackendapi.integration
+
+import com.cag.cagbackendapi.constants.DetailedErrorMessages
+import com.cag.cagbackendapi.constants.RestErrorMessages
+import com.cag.cagbackendapi.error.ErrorDetails
+import com.cag.cagbackendapi.error.exceptions.UnauthorizedException
+import com.cag.cagbackendapi.model.UserModel
+import com.cag.cagbackendapi.util.SpringCommandLineProfileResolver
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.*
+import org.springframework.test.context.ActiveProfiles
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles(value = ["LOCAL"], resolver = SpringCommandLineProfileResolver::class)
+class UserControllerIntegrationTests {
+
+    @Autowired
+    lateinit var testRestTemplate: TestRestTemplate
+
+    private val objectMapper = ObjectMapper()
+
+    @Test
+    fun registerUser_validInput_201Success() {
+        val testUser = UserModel("john testy", "jj@aol.com")
+
+        val headers = HttpHeaders()
+        headers.set("authKey", "mockAuthKey")
+        val request = HttpEntity(testUser, headers)
```

Comment:
I looked it up but I am still unsure on what HttpEntity does, and what an Http request is suppose to do.

Reply:
HttpEntity is a class that can be used to create a request we can send to our controller. This is identical to a request a user could make in Postman to some endpoint. The request has a body and headers. The request body is used to store data we want to send that we don't need to hide, and the headers include data we want to send that we do want to hide like an authKey or other secret values.